ios/PromotedMetricsModule.swift

```diff
@@ -265,32 +265,47 @@ public extension PromotedMetricsModule {
     }
     logger.collectionViewDidChangeVisibleContent(contentList)
   }
-  
+
   /// Ends tracking session for given collection view.
   /// Drops all associated impression logging state.
   ///
   /// - Parameter collectionViewName: Identifier for collection view to track.
-  @objc(collectionViewDidUnmount:)
-  func collectionViewDidUnmount(collectionViewName: String) {
+  @objc(collectionViewWillUnmount:)
+  func collectionViewWillUnmount(collectionViewName: String) {
     if let logger = nameToImpressionLogger.removeValue(forKey: collectionViewName) {
       logger.collectionViewDidHideAllContent()
     }
   }
 }
 
-// MARK: - Session information
+// MARK: - Ancestor IDs
 public extension PromotedMetricsModule {
-  @objc(getLoggingSessionInfo:rejecter:)
-  func loggingSessionInfo(resolver: RCTPromiseResolveBlock, rejecter: RCTPromiseRejectBlock) {
+  @objc(getCurrentOrPendingAncestorIds:rejecter:)
+  func currentOrPendingAncestorIDs(resolver: RCTPromiseResolveBlock,
+                                   rejecter: RCTPromiseRejectBlock) {
     guard let metricsLogger = metricsLogger else {
       resolver([:])
       return
     }
     let sessionInfo = [
-      "logUserId": metricsLogger.logUserID,
-      "sessionId": metricsLogger.sessionID,
-      "viewId": metricsLogger.viewID,
+      "logUserId": metricsLogger.currentOrPendingLogUserID,
```

Comment:
not a blocker - this is the type of scoping that I was expecting for these fields.  When I think about logUserId - I think about 'currentOrPending' version of it.  If this is the default, we can rename the normal field ('logUserID') to this behavior.

Reply:
What would we call the other version of the field?